src/data/data2d/Spectrum2D/contours.ts

```diff
@@ -43,105 +44,127 @@ interface ReturnContoursManager {
   checkLevel: () => Level;
 }
 
-function getDefaultContoursLevel(options: ContourOptions) {
-  const defaultLevel: Level = { negative: 10, positive: 10 };
-  for (const sign of LEVEL_SIGNS) {
-    const [min, max] = options[sign].contourLevels;
-    defaultLevel[sign] = min + max / 2;
-  }
+function getDefaultContoursLevel(spectrum: Spectrum2D, quadrant = 'rr') {
+  const { data, info } = spectrum;
+
+  const quadrantData = data[quadrant];
+
+  //@ts-expect-error will be included in nexts versions
+  const { noise = calculateSanPlot('2D', quadrantData) } = info;
+
+  const { positive, negative } = noise;
+
+  const max = Math.max(
+    Math.abs(quadrantData.minZ),
+    Math.abs(quadrantData.maxZ),
+  );
+
+  const minLevel = Math.ceil(
+    10 *
+      Math.log2(
+        1 + (3 * xMaxAbsoluteValue([positive, negative]) * (2 ** 10 - 1)) / max,
+      ),
+  );
```

Comment:
This calculation needs a comment or to be moved to a function that has a descriptive name and jsdoc.

Reply:
it is the inverse function of max * (2 ** (level / 10) - 1)) / (2 ** 10 - 1).